src/BpmnJSTracking.js

```diff
@@ -1,5 +1,50 @@
-function BpmnJSTracking() {}
+import trackingModules from './trackingModules';
+class BpmnJSTracking {
+
+  constructor(config, selection) {
+    this._config = config;
+    this._isEnabled = false;
+
+    this.trackingModules = [];
+
+    trackingModules.forEach((trackingConstructor) => {
+      this.trackingModules.push(new trackingConstructor({
+        track: this.sendEvent.bind(this),
+        _selection: selection
+      }));
+    });
+  }
+
+  sendEvent(event, payload) {
+    if (this.isEnabled()) {
+      this._config.track(event, payload);
+    }
+  }
+
+  isEnabled() {
+    return this._isEnabled;
+  }
+
+  enable() {
+    this._config.optIn();
+    this._isEnabled = true;
+
+    this.trackingModules.forEach((module) => {
+      module.enable();
+    });
+  }
+
+  disable() {
```

Comment:
Should also unmount / disable on `diagram.destroy`?

That obviously depends on the usage.